lib/token.js

```diff
@@ -379,26 +387,66 @@ function getIdToken(sdk, oauthOptions, options) {
   function handleOAuthResponse(res) {
     if (res['error'] || res['error_description']) {
       throw new OAuthError(res['error'], res['error_description']);
+    }
 
-    } else if (res['id_token']) {
-      if (res.state !== oauthParams.state) {
-        throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
-      }
+    if (res.state !== oauthParams.state) {
+      throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
+    }
+
+    var tokenTypes = oauthParams.responseType;
+    var scopes = oauthQueryHash.scope.split(' ');
+    var tokenDict = {};
 
+    if (res['id_token']) {
```

Comment:
Can we add some comment at the top of this function or somewhere in the code where we document the response types we're returning? I.e. something along the lines of "if we get an array of tokenTypes, we return an array, else we just return the one token that was asked for"